### ISSUE TITLE
benchmark.data_structures fails(bus error) on mimxrt1020/60/64/frdmk64f  platform

### ISSUE THREAD

**Issue by hakehuang** [zephyrproject-rtos/zephyr#28157]
**Describe the bug**
meed bus error when do rb testing

**To Reproduce**
cd  to tests/benchmarks/data_structure_perf/rbtree_perf
1. mkdir build; cd build
2. cmake -DBOARD=mimxrt1020_evk ..
3. make

mimxrt1060_evk and mimxrt1064_evk and mimx8qm_m4 have the same problem, but 1050/1015/1010 are ok.

frdmk64f has the same problem

**Expected behavior**
test pass

**Impact**
rbtree performance

**Logs and console output**

```
start stack at 0x800019c0 !!!
*** Booting Zephyr OS version 2.4.0-rc1  ***

Running test suite rbtree

===================================================================

START - test_rbtree_container

E: ***** BUS FAULT *****

E:   Imprecise data bus error

E: ***** HARD FAULT *****

E:   Fault escalation (see below)

E: ***** BUS FAULT *****

E:   Precise data bus error

E:   BFAR Address: 0xbd3f92c0

E: r0/a1:  0x80002320  r1/a2:  0xbd3f92c0  r2/a3:  0x00000020

E: r3/a4:  0x80002320 r12/ip:  0xb060f0f8 r14/lr:  0x6000385f

E:  xpsr:  0x81000205

E: Faulting instruction address (r15/pc): 0x6000631a

E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0

E: Fault during interrupt handling



E: Current thread: 0x800000e8 (unknown)

E: Halting system
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux,)
 - Toolchain ( Zephyr SDK, ...)
 - Commit SHA or Version used: v2.4.0-rc1

**Comment by hakehuang** [comment]
@MaureenHelm @dleach02

**Comment by MaureenHelm** [comment]
Fixed by #28140
